### PR TITLE
HPCC-19847 Unicode library version check and c++ standard change

### PR DESCRIPTION
**Pull request by Michael-Gardner**
Signed-off-by: Michael Gardner <michael.gardner@lexisnexisrisk.net>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [x] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by Michael-Gardner** [comment]
@xwang2713 Please review.  Commit message of course will change.  Jira seems to be down and I'll update the PR as soon as I can view the ticket number to assign it.

**Review by xwang2713 — COMMENTED** [review]
Looks good

**Comment by Michael-Gardner** [comment]
@ghalliday Ready for merge

File: rtl/eclrtl/CMakeLists.txt [hunk 1]
```diff
@@ -80,7 +80,8 @@ include_directories (
 ADD_DEFINITIONS( -D_USRDLL -DECLRTL_EXPORTS )
 
 if (USE_BOOST_REGEX)
-    if ((CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_CLANGXX) AND (${BOOST_REGEX_VERSION} LESS 105700) AND NOT ICU_REQUIRES_CPP11)
```
Comment (richardkchapman) [inline rtl/eclrtl/CMakeLists.txt]: Should we remove the ICU_REQUIRES_CPP11 option now - I think this is the only usage of it?

**Comment by richardkchapman** [comment]
@Michael-Gardner One question

File: rtl/eclrtl/CMakeLists.txt (hunk 1 above)
Comment (Michael-Gardner) [reply rtl/eclrtl/CMakeLists.txt]: Yeah I'll go ahead and remove it since it's unused now

Comment (Michael-Gardner) [reply rtl/eclrtl/CMakeLists.txt]: @richardkchapman that option has been removed and this PR is ready for merge.

**Pull request merged**